site.ts

```diff
@@ -220,22 +221,38 @@ export function inject({ config, posthog }) {
         })
     }
 
-    posthog?.getActiveMatchingSurveys((surveys) => {
-        surveys.forEach((survey) => {
-            const surveyName = survey.name.replace(/\s/g , "-")
-            if (!localStorage.getItem(`seenSurvey_${surveyName}_${survey.id}`)) {
-                const shadow = createShadow(style(surveyName, survey?.appearance))
-                const surveyPopup = createSurveyPopup(survey, surveyName)
-                addListeners(surveyPopup, surveyName, survey.id, survey.name)
-                shadow.appendChild(surveyPopup)
-                window.dispatchEvent(new Event('PHSurveyShown'))
-                const sessionRecordingUrl = getSessionRecordingUrl()
-                posthog.capture('survey shown', {
-                    $survey_name: survey.name,
-                    $survey_id: survey.id,
-                    sessionRecordingUrl: sessionRecordingUrl,
-                })
+    const callSurveys = (posthog) => {
+        posthog?.getActiveMatchingSurveys((surveys) => {
+            surveys.forEach((survey) => {
+                if (document.getElementsByClassName(`PostHogSurvey${survey.id}`).length == 0) {
+                    const surveyName = survey.name.replace(/\s/g, "-")
+                    if (!localStorage.getItem(`seenSurvey_${surveyName}_${survey.id}`)) {
+                        const shadow = createShadow(style(surveyName, survey?.appearance), survey.id)
+                        const surveyPopup = createSurveyPopup(survey, surveyName)
+                        addListeners(surveyPopup, surveyName, survey.id, survey.name)
+                        shadow.appendChild(surveyPopup)
+                        window.dispatchEvent(new Event('PHSurveyShown'))
+                        const sessionRecordingUrl = getSessionRecordingUrl()
+                        posthog.capture('survey shown', {
+                            $survey_name: survey.name,
+                            $survey_id: survey.id,
+                            sessionRecordingUrl: sessionRecordingUrl,
+                        })
+                    }
+                }
+            })
+        }, true)
+    }
+
+    callSurveys(posthog)
+
+    let currentUrl = location.href
+    if (location.href) {
+        setInterval(() => {
```

Comment:
'popstate' and 'hashchange' event listeners don't really work here so we are using setInterval